contracts/vaults/Deposit.sol

```diff
@@ -97,6 +96,7 @@ contract Deposit is Base {
         // if the user already has funds in the pool, give the previous reward
         if (_user.shares > 0) {
             uint256 pending = _user.shares * poolInfos[_pid].rewardPerShare / 1e12 - _user.rewardDebt;
+            userInfo[_pid][msg.sender].rewardDebt += pending;
```

Comment:
this works but it is a very unclear way of writing 

`user.rewardDebt = user.shares * pool.rewardPerShare / 1e12;`

In any case, I guess the update of the rewardDebt should not be part of the `_claimReward` function, bc (see my other comment) it is just useless when we call `_claimReward` as part fo the deposit function

Reply:
Ok I just removed the _claimReward function and made the calculation clear like before